components/mod-event/useModEventList.tsx

```diff
@@ -225,6 +222,42 @@ export const useModEventList = (
         queryParams.addedTags = removedTags.trim().split(',')
       }
 
+      const filterTypes = types.filter(Boolean)
+      if (filterTypes.length < allTypes.length && filterTypes.length > 0) {
+        queryParams.types = filterTypes.map((type) => {
+          // There is a no appeal type, it's a placeholder and behind the scene
+          // we use type as report and reportTypes as appeal
+          if (type === MOD_EVENTS.APPEAL) {
+            if (!queryParams.reportTypes) {
+              queryParams.reportTypes = []
+            }
+            queryParams.reportTypes.push(ComAtprotoModerationDefs.REASONAPPEAL)
+            return MOD_EVENTS.REPORT
+          }
+
+          // We use custom event type name that translate to either add or remove certain tag
+          const { add, remove } = buildTagFilter(type)
+          if (add.length || remove.length) {
+            if (add.length) {
+              if (queryParams.addedTags) {
+                queryParams.addedTags.push(...add)
+              } else {
+                queryParams.addedTags = add
+              }
```

Comment:
Nit: the `||=` notation make things easier to read imo

```suggestion
              queryParams.addedTags ||= []
              queryParams.addedTags.push(...add)
```